scripts/haddocks.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+
+# Build haddock documentation and an index for all projects in
+# `cardano-base` repository.
+#
+# usage:
+# ./haddocks.sh directory [true|false]
+#
+# $1 - where to put the generated pages, this directory contents will be wiped
+#      out (so don't pass `/` or `./` - the latter will delete your 'dist-newstyle')
+#      (the default is './haddocks')
+# $2 - whether to re-build haddocks with `cabal haddock` command or a component name
+#      (the default is true)
+#
+
+set -euo pipefail
+
+OUTPUT_DIR=${1:-"./haddocks"}
+REGENERATE=${2:-"true"}
+
+BUILD_DIR="dist-newstyle"
+GHC_VERSION=$(ghc --numeric-version)
+OS_ARCH="$(cat dist-newstyle/cache/plan.json | jq -r '.arch + "-" + .os' | head -n 1 | xargs)"
+
+
+# Generate  `doc-index.json` and `doc-index.html` per package, to assemble them later at the top level.
+HADDOCK_OPTS=(
+    --builddir "${BUILD_DIR}"
+    --haddock-all
+    --haddock-internal
+    --haddock-html
+    --haddock-quickjump
+    --haddock-hyperlink-source
+    --haddock-option "--show-all"
+    --haddock-option "--use-unicode"
+    --haddock-option="--base-url=.."
+  )
+
+# build documentation of all modules
+if [ ${REGENERATE} == "true" ]; then
+  cabal haddock "${HADDOCK_OPTS[@]}" cardano-binary-test
```

Comment:
Is `cardano-binary-test` special in some way? Shouldn't it be:
```suggestion
  cabal haddock "${HADDOCK_OPTS[@]}" all
```

Reply:
I left this by accident after testing something!! Thank you for spotting it